conans/test/functional/toolchains/gnu/test_makedeps.py

```diff
@@ -0,0 +1,119 @@
+    client = TestClient(path_with_spaces=False)
+    with client.chdir("lib"):
+        client.run("new cmake_lib -d name=app -d version=1.0")
+        client.run("create . -tf=\"\"")
+    with client.chdir("test"):
+        client.run("new cmake_lib -d name=test -d version=1.0")
+        client.run("create . -tf=\"\"")
+    with client.chdir("tool"):
+        client.run("new cmake_lib -d name=tool -d version=1.0")
+        client.run("create . -tf=\"\"")
+    # Create library having build and test requires
+    conanfile = textwrap.dedent(r'''
+        from conan import ConanFile
+        class HelloLib(ConanFile):
+            def build_requirements(self):
+                self.test_requires('app/1.0')
+                self.test_requires('test/1.0')
+                self.tool_requires('tool/1.0')
+        ''')
+    client.save({"conanfile.py": conanfile}, clean_first=True)
+    client.run("install . -g MakeDeps")
+    content = client.load(CONAN_MAKEFILE_FILENAME)
+    assert "CONAN_NAME_TEST" in content
+    assert "CONAN_NAME_APP" in content
+
+
+@pytest.mark.tool("msys2")
+@pytest.mark.tool("mingw64")
+def test_makedeps_with_makefile_build():
+    """
+    Build a small application using MakeDeps generator
+    """
+    client = TestClient(path_with_spaces=False)
+    with client.chdir("lib"):
+        client.save({"Makefile": gen_makefile(libs=["hello"]),
+                    "hello.cpp": gen_function_cpp(name="hello", includes=["hello"], calls=["hello"]),
+                    "hello.h": gen_function_h(name="hello"),
+                    "conanfile.py": textwrap.dedent(r'''
+                    from conan import ConanFile
+                    from conan.tools.gnu import Autotools
+                    from conan.tools.files import copy
+                    import os
+                    class PackageConan(ConanFile):
+                        generators = "AutotoolsToolchain"
+                        exports_sources = ("Makefile", "hello.cpp", "hello.h")
+                        settings = "os", "arch", "compiler"
+
+                        def configure(self):
+                            self.win_bash = self.settings.os == "Windows"
+
+                        def build(self):
+                            Autotools(self).make()
+
+                        def package(self):
+                            copy(self, "libhello.a", src=self.build_folder, dst=os.path.join(self.package_folder, "lib"))
+                            copy(self, "hello.h", src=self.source_folder, dst=os.path.join(self.package_folder, "include"))
+
+                        def package_info(self):
+                            self.cpp_info.libs = ["hello"]
+                    ''')
+                     })
+        client.run('create . --name=hello --version=0.1.0 -c tools.microsoft.bash:subsystem=msys2 -c tools.microsoft.bash:path=bash')
+    with client.chdir("app"):
+        client.run("install --requires=hello/0.1.0 -pr:b=default -pr:h=default -g MakeDeps -of build")
```

Comment:
But if ``MakeDeps`` is used only in the consumer side, why the complexity of building the package with Autotools, that is slow and complicated? It should work with a standard template recipe, even cmake_lib